webapp/components/survey/CategorySelector/CategorySelector.js

```diff
@@ -4,22 +4,20 @@ import React from 'react'
 import PropTypes from 'prop-types'
 
 import { useDispatch } from 'react-redux'
-import { Link, useHistory } from 'react-router-dom'
+import { useHistory } from 'react-router-dom'
 
 import * as Survey from '@core/survey/survey'
 import * as Category from '@core/survey/category'
 
 import { useI18n } from '@webapp/store/system'
 import Dropdown from '@webapp/components/form/dropdown'
 
-import { appModuleUri, designerModules, analysisModules } from '@webapp/app/appModules'
-
 import { useSurvey } from '@webapp/store/survey'
 
-import { createCategory } from '@webapp/loggedin/surveyViews/category/actions'
+import { createCategory, navigateToCategoriesManager } from '@webapp/loggedin/surveyViews/category/actions'
 
 const CategorySelector = (props) => {
-  const { disabled, categoryUuid, validation, showManage, showAdd, analysis, onChange } = props
+  const { disabled, categoryUuid, validation, showManage, showAdd, analysis, nodeDefState, onChange } = props
```

Comment:
why do you pass the nodeDefState here? can't a nodeDefUuid in url param be enough ?